workers/data_refinery_workers/processors/smasher.py

```diff
@@ -164,15 +164,16 @@ def _smash(job_context: Dict) -> Dict:
         metadata['files'] = os.listdir(smash_path)
         # Metadata to JSON
         metadata['created_at'] = datetime.utcnow().strftime('%Y-%m-%dT%H:%M:%S')
-        with open(smash_path + 'metadata.json', 'w') as metadata_file: 
+        with open(smash_path + 'metadata.json', 'w') as metadata_file:
             json.dump(metadata, metadata_file, indent=4, sort_keys=True)
 
         # Finally, compress all files into a zip
-        final_zip = smash_path + str(job_context['dataset'].id)
-        shutil.make_archive(final_zip, 'zip', smash_path)
-        job_context["output_file"] = final_zip + ".zip"
+        shutil.make_archive(smash_path, 'zip', smash_path)
+        job_context["output_file"] = smash_path + ".zip"
+        # and clean up the unzipped directory.
+        shutil.rmtree(smash_path)
```

Comment:
Surely this will destroy the file we just created?

Also, `smash_path = "/home/user/data_store/smashed/" + str(job_context["dataset"].pk) + "/" + ".zip"`?

This doesn't look right to me

Reply:
`smash_path = "/home/user/data_store/smashed/" + str(job_context["dataset"].pk) + "/" + ".zip"` You're right about this. I need to pop off that trailing slash. However once I do the zip will be at `/home/user/data_store/smashed/<DATASET_ID>.zip` and we'll delete `/home/user/data_store/smashed/<DATASET_ID>/` so our zip will survive.